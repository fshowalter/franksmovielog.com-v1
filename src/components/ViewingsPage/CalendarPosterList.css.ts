import { style } from "@vanilla-extract/css";
import { minMediaQuery } from "../../styles/breakpoints";

export const slugTypographyStyle = style({
  fontSize: ".875rem",
  lineHeight: "1rem",
  letterSpacing: "0.5px",
});

export const titleTypographyStyle = style({
  fontSize: "1rem",
  lineHeight: "1.5rem",
});

export const posterStyle = style({
  borderRadius: "4px",
  maxWidth: "64px",
  minWidth: "64px",

  "@media": {
    [minMediaQuery("tablet")]: {
      // maxWidth: "80px",
      // minWidth: "80px",
    },
  },
});

export const listItemStyle = style({
  ":last-of-type": {
    boxShadow: "none",
  },
});
