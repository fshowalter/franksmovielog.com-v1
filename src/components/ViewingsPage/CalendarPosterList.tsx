import { graphql } from "gatsby";
import { Box, IBoxProps } from "../Box";
import { GraphqlImage, IGraphqlImage } from "../GraphqlImage";
import { Link } from "../Link";
import { Spacer } from "../Spacer";
import {
  listItemStyle,
  posterStyle,
  slugTypographyStyle,
  titleTypographyStyle,
} from "./CalendarPosterList.css";

function MediumAndVenue({
  medium,
  venue,
}: {
  medium?: string | null;
  venue?: string | null;
}): JSX.Element | null {
  if (medium && venue) {
    return (
      <>
        <div>
          {medium} at {venue}
        </div>
      </>
    );
  }

  if (medium) {
    return (
      <>
        <div>{medium}</div>
      </>
    );
  }

  if (venue) {
    return (
      <>
        <div>{venue}</div>
      </>
    );
  }

  return null;
}

interface IImageProps extends IBoxProps {
  slug: string | null | undefined;
  image: IGraphqlImage;
  title: string;
  year: number;
}

function Image({ slug, image, title, year, ...rest }: IImageProps) {
  if (slug) {
    return (
      <Link
        className={posterStyle}
        overflow="hidden"
        to={`/reviews/${slug}/`}
        transform="safariBorderRadiusFix"
        {...rest}
      >
        <GraphqlImage image={image} alt={`A poster from ${title} (${year})`} />
      </Link>
    );
  }

  return (
    <GraphqlImage
      image={image}
      alt="An unreviewed title."
      className={posterStyle}
      overflow="hidden"
      transform="safariBorderRadiusFix"
    />
  );
}

function Title({
  title,
  year,
  slug,
}: {
  title: string;
  year: number;
  slug: string | null | undefined;
}) {
  const yearBox = (
    <Box as="span" fontSize="xSmall" color="subtle" fontWeight="light">
      {year}
    </Box>
  );

  if (slug)
    return (
      <Link
        to={`/reviews/${slug}/`}
        className={titleTypographyStyle}
        display="block"
      >
        {title}&#8239;&#8239;{yearBox}
      </Link>
    );

  return (
    <Box className={titleTypographyStyle}>
      {title}&#8239;&#8239;{yearBox}
    </Box>
  );
}

export function CalendarPoster({
  slug,
  image,
  title,
  year,
  venue,
  medium,
}: {
  slug?: string | null;
  image: IGraphqlImage;
  title: string;
  year: number;
  venue?: string | null;
  medium?: string | null;
}): JSX.Element {
  return (
    <Box
      as="li"
      flexDirection="row"
      columnGap={{ default: 16, tablet: 24 }}
      alignItems="flex-start"
      display="flex"
      paddingRight="gutter"
      boxShadow="borderBottom"
      paddingBottom={16}
      className={listItemStyle}
    >
      <Image
        slug={slug}
        image={image}
        title={title}
        year={year}
        flexShrink={0}
        boxShadow="borderAll"
      />
      <Box flexGrow={1}>
        <Box>
          <Title title={title} year={year} slug={slug} />
          <Spacer axis="vertical" size={{ default: 4, tablet: 8 }} />
        </Box>
        <Box
          color="subtle"
          display="flex"
          flexDirection="column"
          className={slugTypographyStyle}
          fontWeight="light"
          letterSpacing={0.5}
        >
          <Spacer axis="vertical" size={{ default: 4, tablet: 0 }} />
          <Box>
            <MediumAndVenue medium={medium} venue={venue} />
          </Box>
        </Box>
      </Box>
    </Box>
  );
}

export function CalendarPosterList({
  children,
  ...rest
}: IBoxProps): JSX.Element {
  return (
    <Box as="ol" paddingX={0} {...rest}>
      {children}
    </Box>
  );
}

export const query = graphql`
  fragment CalendarPoster on File {
    childImageSharp {
      gatsbyImageData(
        layout: CONSTRAINED
        formats: [JPG, AVIF]
        quality: 80
        width: 80
        placeholder: NONE
      )
    }
  }
`;
