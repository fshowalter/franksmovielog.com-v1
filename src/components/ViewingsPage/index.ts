import ViewingsPage from "./ViewingsPage";

export default ViewingsPage;
