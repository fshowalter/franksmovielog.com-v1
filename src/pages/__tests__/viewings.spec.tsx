// eslint-disable-next-line import/no-extraneous-dependencies
import { act, render, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import React from "react";
import ViewingsPage from "../viewings";
import data from "./__fixtures__/viewings-page-queries";

jest.mock("../../components/Seo/Seo.tsx");

describe("/viewings", () => {
  it("renders", () => {
    const { asFragment } = render(<ViewingsPage data={data} />);

    expect(asFragment()).toMatchSnapshot();
  });

  // Helmet uses requestAnimationFrame to ensure DOM is synced.
  // https://github.com/nfl/react-helmet/blob/master/test/HelmetDeclarativeTest.js
  // eslint-disable-next-line jest/no-done-callback
  it("sets page title", (done) => {
    expect.hasAssertions();
    render(<ViewingsPage data={data} />);

    requestAnimationFrame(() => {
      expect(document.title).toStrictEqual("Viewing Log");
      done();
    });
  });

  it("can filter by title", async () => {
    expect.hasAssertions();
    render(<ViewingsPage data={data} />);

    act(() => {
      jest.useFakeTimers(); // For the debouced input
      userEvent.type(screen.getByLabelText("Title"), "Human Tornado");
      jest.runOnlyPendingTimers(); // Flush the delay
      jest.useRealTimers();
    });

    await waitFor(() => {
      expect(screen.getByTestId("viewings-list")).toMatchSnapshot();
    });
  });

  it("can filter by venue", () => {
    expect.hasAssertions();
    render(<ViewingsPage data={data} />);

    userEvent.selectOptions(screen.getByLabelText("Venue"), "Arrow Player");

    expect(screen.getByTestId("viewings-list")).toMatchSnapshot();
  });

  it("can filter by venue then show all", () => {
    render(<ViewingsPage data={data} />);

    userEvent.selectOptions(screen.getByLabelText("Venue"), "Arrow Player");
    userEvent.selectOptions(screen.getByLabelText("Venue"), "All");

    expect(screen.getByTestId("viewings-list")).toMatchSnapshot();
  });

  it("can sort by viewing date with newest first", () => {
    render(<ViewingsPage data={data} />);

    userEvent.selectOptions(
      screen.getByLabelText("Order By"),
      "Viewing Date (Newest First)"
    );

    expect(screen.getByTestId("viewings-list")).toMatchSnapshot();
  });

  it("can sort by viewing date with oldest first", () => {
    render(<ViewingsPage data={data} />);

    userEvent.selectOptions(
      screen.getByLabelText("Order By"),
      "Viewing Date (Oldest First)"
    );

    expect(screen.getByTestId("viewings-list")).toMatchSnapshot();
  });

  it("can sort by title", () => {
    render(<ViewingsPage data={data} />);

    userEvent.selectOptions(screen.getByLabelText("Order By"), "Title");

    expect(screen.getByTestId("viewings-list")).toMatchSnapshot();
  });

  it("can sort by release date with oldest first", () => {
    render(<ViewingsPage data={data} />);

    userEvent.selectOptions(
      screen.getByLabelText("Order By"),
      "Release Date (Oldest First)"
    );

    expect(screen.getByTestId("viewings-list")).toMatchSnapshot();
  });

  it("can sort by release date with newest first", () => {
    render(<ViewingsPage data={data} />);

    userEvent.selectOptions(
      screen.getByLabelText("Order By"),
      "Release Date (Newest First)"
    );

    expect(screen.getByTestId("viewings-list")).toMatchSnapshot();
  });

  it("can filter by release year", () => {
    render(<ViewingsPage data={data} />);

    const inputs = screen.getAllByLabelText("Release Year");

    userEvent.clear(inputs[0]);
    userEvent.type(inputs[0], "1980");
    userEvent.clear(inputs[1]);
    userEvent.type(inputs[1], "2010");

    expect(screen.getByTestId("viewings-list")).toMatchSnapshot();
  });
});
