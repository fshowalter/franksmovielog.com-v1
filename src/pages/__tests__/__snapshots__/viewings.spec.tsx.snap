// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/viewings can filter by release year 1`] = `
<ol
  class="listCss"
  data-testid="viewings-list"
>
  <li
    class="listItemCss listItemFirstCss"
    value="1113"
  >
    <div
      class="listItemTitleCss"
    >
      <a
        class="listItemTitleLinkCss"
        href="/reviews/weekend-at-bernies-1989/#1113"
        rel="canonical"
      >
        Weekend at Bernie's
         
        <span
          class="listItemTitleYearCss"
        >
          1989
        </span>
      </a>
    </div>
    <div
      class="listItemSlugCss"
    >
      Saturday Jun 12, 2021
       via 
      Hulu
    </div>
  </li>
  <li
    class="listItemCss "
    value="1111"
  >
    <div
      class="listItemTitleCss"
    >
      <a
        class="listItemTitleLinkCss"
        href="/reviews/gone-with-the-pope-2010/#1111"
        rel="canonical"
      >
        Gone with the Pope
         
        <span
          class="listItemTitleYearCss"
        >
          2010
        </span>
      </a>
    </div>
    <div
      class="listItemSlugCss"
    >
      Thursday Jun 10, 2021
       via 
      Arrow Player
    </div>
  </li>
</ol>
`;

exports[`/viewings can filter by title 1`] = `
<ol
  class="listCss"
  data-testid="viewings-list"
>
  <li
    class="listItemCss listItemFirstCss"
    value="1112"
  >
    <div
      class="listItemTitleCss"
    >
      <a
        class="listItemTitleLinkCss"
        href="/reviews/the-human-tornado-1976/#1112"
        rel="canonical"
      >
        The Human Tornado
         
        <span
          class="listItemTitleYearCss"
        >
          1976
        </span>
      </a>
    </div>
    <div
      class="listItemSlugCss"
    >
      Thursday Jun 10, 2021
       via 
      Blu-ray
    </div>
  </li>
</ol>
`;

exports[`/viewings can filter by venue 1`] = `
<ol
  class="listCss"
  data-testid="viewings-list"
>
  <li
    class="listItemCss listItemFirstCss"
    value="1111"
  >
    <div
      class="listItemTitleCss"
    >
      <a
        class="listItemTitleLinkCss"
        href="/reviews/gone-with-the-pope-2010/#1111"
        rel="canonical"
      >
        Gone with the Pope
         
        <span
          class="listItemTitleYearCss"
        >
          2010
        </span>
      </a>
    </div>
    <div
      class="listItemSlugCss"
    >
      Thursday Jun 10, 2021
       via 
      Arrow Player
    </div>
  </li>
</ol>
`;

exports[`/viewings can filter by venue then show all 1`] = `
<ol
  class="listCss"
  data-testid="viewings-list"
>
  <li
    class="listItemCss listItemFirstCss"
    value="1113"
  >
    <div
      class="listItemTitleCss"
    >
      <a
        class="listItemTitleLinkCss"
        href="/reviews/weekend-at-bernies-1989/#1113"
        rel="canonical"
      >
        Weekend at Bernie's
         
        <span
          class="listItemTitleYearCss"
        >
          1989
        </span>
      </a>
    </div>
    <div
      class="listItemSlugCss"
    >
      Saturday Jun 12, 2021
       via 
      Hulu
    </div>
  </li>
  <li
    class="listItemCss "
    value="1112"
  >
    <div
      class="listItemTitleCss"
    >
      <a
        class="listItemTitleLinkCss"
        href="/reviews/the-human-tornado-1976/#1112"
        rel="canonical"
      >
        The Human Tornado
         
        <span
          class="listItemTitleYearCss"
        >
          1976
        </span>
      </a>
    </div>
    <div
      class="listItemSlugCss"
    >
      Thursday Jun 10, 2021
       via 
      Blu-ray
    </div>
  </li>
  <li
    class="listItemCss "
    value="1111"
  >
    <div
      class="listItemTitleCss"
    >
      <a
        class="listItemTitleLinkCss"
        href="/reviews/gone-with-the-pope-2010/#1111"
        rel="canonical"
      >
        Gone with the Pope
         
        <span
          class="listItemTitleYearCss"
        >
          2010
        </span>
      </a>
    </div>
    <div
      class="listItemSlugCss"
    >
      Thursday Jun 10, 2021
       via 
      Arrow Player
    </div>
  </li>
  <li
    class="listItemCss "
    value="1"
  >
    <div
      class="listItemTitleCss"
    >
      The Girl with the Dragon Tattoo
       
      <span
        class="listItemTitleYearCss"
      >
        2011
      </span>
    </div>
    <div
      class="listItemSlugCss"
    >
      Thursday Jan 5, 2012
       via 
      AMC Tysons Corner 16
    </div>
  </li>
</ol>
`;

exports[`/viewings can sort by release date with newest first 1`] = `
<ol
  class="listCss"
  data-testid="viewings-list"
>
  <li
    class="listItemCss listItemFirstCss"
    value="1"
  >
    <div
      class="listItemTitleCss"
    >
      The Girl with the Dragon Tattoo
       
      <span
        class="listItemTitleYearCss"
      >
        2011
      </span>
    </div>
    <div
      class="listItemSlugCss"
    >
      Thursday Jan 5, 2012
       via 
      AMC Tysons Corner 16
    </div>
  </li>
  <li
    class="listItemCss "
    value="1111"
  >
    <div
      class="listItemTitleCss"
    >
      <a
        class="listItemTitleLinkCss"
        href="/reviews/gone-with-the-pope-2010/#1111"
        rel="canonical"
      >
        Gone with the Pope
         
        <span
          class="listItemTitleYearCss"
        >
          2010
        </span>
      </a>
    </div>
    <div
      class="listItemSlugCss"
    >
      Thursday Jun 10, 2021
       via 
      Arrow Player
    </div>
  </li>
  <li
    class="listItemCss "
    value="1113"
  >
    <div
      class="listItemTitleCss"
    >
      <a
        class="listItemTitleLinkCss"
        href="/reviews/weekend-at-bernies-1989/#1113"
        rel="canonical"
      >
        Weekend at Bernie's
         
        <span
          class="listItemTitleYearCss"
        >
          1989
        </span>
      </a>
    </div>
    <div
      class="listItemSlugCss"
    >
      Saturday Jun 12, 2021
       via 
      Hulu
    </div>
  </li>
  <li
    class="listItemCss "
    value="1112"
  >
    <div
      class="listItemTitleCss"
    >
      <a
        class="listItemTitleLinkCss"
        href="/reviews/the-human-tornado-1976/#1112"
        rel="canonical"
      >
        The Human Tornado
         
        <span
          class="listItemTitleYearCss"
        >
          1976
        </span>
      </a>
    </div>
    <div
      class="listItemSlugCss"
    >
      Thursday Jun 10, 2021
       via 
      Blu-ray
    </div>
  </li>
</ol>
`;

exports[`/viewings can sort by release date with oldest first 1`] = `
<ol
  class="listCss"
  data-testid="viewings-list"
>
  <li
    class="listItemCss listItemFirstCss"
    value="1112"
  >
    <div
      class="listItemTitleCss"
    >
      <a
        class="listItemTitleLinkCss"
        href="/reviews/the-human-tornado-1976/#1112"
        rel="canonical"
      >
        The Human Tornado
         
        <span
          class="listItemTitleYearCss"
        >
          1976
        </span>
      </a>
    </div>
    <div
      class="listItemSlugCss"
    >
      Thursday Jun 10, 2021
       via 
      Blu-ray
    </div>
  </li>
  <li
    class="listItemCss "
    value="1113"
  >
    <div
      class="listItemTitleCss"
    >
      <a
        class="listItemTitleLinkCss"
        href="/reviews/weekend-at-bernies-1989/#1113"
        rel="canonical"
      >
        Weekend at Bernie's
         
        <span
          class="listItemTitleYearCss"
        >
          1989
        </span>
      </a>
    </div>
    <div
      class="listItemSlugCss"
    >
      Saturday Jun 12, 2021
       via 
      Hulu
    </div>
  </li>
  <li
    class="listItemCss "
    value="1111"
  >
    <div
      class="listItemTitleCss"
    >
      <a
        class="listItemTitleLinkCss"
        href="/reviews/gone-with-the-pope-2010/#1111"
        rel="canonical"
      >
        Gone with the Pope
         
        <span
          class="listItemTitleYearCss"
        >
          2010
        </span>
      </a>
    </div>
    <div
      class="listItemSlugCss"
    >
      Thursday Jun 10, 2021
       via 
      Arrow Player
    </div>
  </li>
  <li
    class="listItemCss "
    value="1"
  >
    <div
      class="listItemTitleCss"
    >
      The Girl with the Dragon Tattoo
       
      <span
        class="listItemTitleYearCss"
      >
        2011
      </span>
    </div>
    <div
      class="listItemSlugCss"
    >
      Thursday Jan 5, 2012
       via 
      AMC Tysons Corner 16
    </div>
  </li>
</ol>
`;

exports[`/viewings can sort by title 1`] = `
<ol
  class="listCss"
  data-testid="viewings-list"
>
  <li
    class="listItemCss listItemFirstCss"
    value="1"
  >
    <div
      class="listItemTitleCss"
    >
      The Girl with the Dragon Tattoo
       
      <span
        class="listItemTitleYearCss"
      >
        2011
      </span>
    </div>
    <div
      class="listItemSlugCss"
    >
      Thursday Jan 5, 2012
       via 
      AMC Tysons Corner 16
    </div>
  </li>
  <li
    class="listItemCss "
    value="1111"
  >
    <div
      class="listItemTitleCss"
    >
      <a
        class="listItemTitleLinkCss"
        href="/reviews/gone-with-the-pope-2010/#1111"
        rel="canonical"
      >
        Gone with the Pope
         
        <span
          class="listItemTitleYearCss"
        >
          2010
        </span>
      </a>
    </div>
    <div
      class="listItemSlugCss"
    >
      Thursday Jun 10, 2021
       via 
      Arrow Player
    </div>
  </li>
  <li
    class="listItemCss "
    value="1112"
  >
    <div
      class="listItemTitleCss"
    >
      <a
        class="listItemTitleLinkCss"
        href="/reviews/the-human-tornado-1976/#1112"
        rel="canonical"
      >
        The Human Tornado
         
        <span
          class="listItemTitleYearCss"
        >
          1976
        </span>
      </a>
    </div>
    <div
      class="listItemSlugCss"
    >
      Thursday Jun 10, 2021
       via 
      Blu-ray
    </div>
  </li>
  <li
    class="listItemCss "
    value="1113"
  >
    <div
      class="listItemTitleCss"
    >
      <a
        class="listItemTitleLinkCss"
        href="/reviews/weekend-at-bernies-1989/#1113"
        rel="canonical"
      >
        Weekend at Bernie's
         
        <span
          class="listItemTitleYearCss"
        >
          1989
        </span>
      </a>
    </div>
    <div
      class="listItemSlugCss"
    >
      Saturday Jun 12, 2021
       via 
      Hulu
    </div>
  </li>
</ol>
`;

exports[`/viewings can sort by viewing date with newest first 1`] = `
<ol
  class="listCss"
  data-testid="viewings-list"
>
  <li
    class="listItemCss listItemFirstCss"
    value="1113"
  >
    <div
      class="listItemTitleCss"
    >
      <a
        class="listItemTitleLinkCss"
        href="/reviews/weekend-at-bernies-1989/#1113"
        rel="canonical"
      >
        Weekend at Bernie's
         
        <span
          class="listItemTitleYearCss"
        >
          1989
        </span>
      </a>
    </div>
    <div
      class="listItemSlugCss"
    >
      Saturday Jun 12, 2021
       via 
      Hulu
    </div>
  </li>
  <li
    class="listItemCss "
    value="1112"
  >
    <div
      class="listItemTitleCss"
    >
      <a
        class="listItemTitleLinkCss"
        href="/reviews/the-human-tornado-1976/#1112"
        rel="canonical"
      >
        The Human Tornado
         
        <span
          class="listItemTitleYearCss"
        >
          1976
        </span>
      </a>
    </div>
    <div
      class="listItemSlugCss"
    >
      Thursday Jun 10, 2021
       via 
      Blu-ray
    </div>
  </li>
  <li
    class="listItemCss "
    value="1111"
  >
    <div
      class="listItemTitleCss"
    >
      <a
        class="listItemTitleLinkCss"
        href="/reviews/gone-with-the-pope-2010/#1111"
        rel="canonical"
      >
        Gone with the Pope
         
        <span
          class="listItemTitleYearCss"
        >
          2010
        </span>
      </a>
    </div>
    <div
      class="listItemSlugCss"
    >
      Thursday Jun 10, 2021
       via 
      Arrow Player
    </div>
  </li>
  <li
    class="listItemCss "
    value="1"
  >
    <div
      class="listItemTitleCss"
    >
      The Girl with the Dragon Tattoo
       
      <span
        class="listItemTitleYearCss"
      >
        2011
      </span>
    </div>
    <div
      class="listItemSlugCss"
    >
      Thursday Jan 5, 2012
       via 
      AMC Tysons Corner 16
    </div>
  </li>
</ol>
`;

exports[`/viewings can sort by viewing date with oldest first 1`] = `
<ol
  class="listCss"
  data-testid="viewings-list"
>
  <li
    class="listItemCss listItemFirstCss"
    value="1"
  >
    <div
      class="listItemTitleCss"
    >
      The Girl with the Dragon Tattoo
       
      <span
        class="listItemTitleYearCss"
      >
        2011
      </span>
    </div>
    <div
      class="listItemSlugCss"
    >
      Thursday Jan 5, 2012
       via 
      AMC Tysons Corner 16
    </div>
  </li>
  <li
    class="listItemCss "
    value="1111"
  >
    <div
      class="listItemTitleCss"
    >
      <a
        class="listItemTitleLinkCss"
        href="/reviews/gone-with-the-pope-2010/#1111"
        rel="canonical"
      >
        Gone with the Pope
         
        <span
          class="listItemTitleYearCss"
        >
          2010
        </span>
      </a>
    </div>
    <div
      class="listItemSlugCss"
    >
      Thursday Jun 10, 2021
       via 
      Arrow Player
    </div>
  </li>
  <li
    class="listItemCss "
    value="1112"
  >
    <div
      class="listItemTitleCss"
    >
      <a
        class="listItemTitleLinkCss"
        href="/reviews/the-human-tornado-1976/#1112"
        rel="canonical"
      >
        The Human Tornado
         
        <span
          class="listItemTitleYearCss"
        >
          1976
        </span>
      </a>
    </div>
    <div
      class="listItemSlugCss"
    >
      Thursday Jun 10, 2021
       via 
      Blu-ray
    </div>
  </li>
  <li
    class="listItemCss "
    value="1113"
  >
    <div
      class="listItemTitleCss"
    >
      <a
        class="listItemTitleLinkCss"
        href="/reviews/weekend-at-bernies-1989/#1113"
        rel="canonical"
      >
        Weekend at Bernie's
         
        <span
          class="listItemTitleYearCss"
        >
          1989
        </span>
      </a>
    </div>
    <div
      class="listItemSlugCss"
    >
      Saturday Jun 12, 2021
       via 
      Hulu
    </div>
  </li>
</ol>
`;

exports[`/viewings renders 1`] = `
<DocumentFragment>
  <a
    class="skipLinkCss"
    href="#content"
  >
    Skip to content
  </a>
  <div
    class="containerCss"
    id="top"
  >
    <header
      class="mastHeaderCss"
    >
      <h1
        class="mastTitleCss"
      >
        <a
          href="/"
        >
          Frank's Movie Log
        </a>
      </h1>
      <p
        class="mastTaglineCss"
      >
        My life at the movies.
      </p>
      <form
        accept-charset="UTF-8"
        action="https://www.google.com/search"
        class="mastSearchFormCss"
        method="get"
        role="search"
      >
        <label
          class="mastSearchWrapCss"
          for="search"
        >
          <div
            class="srOnlyCss"
          >
            Search
          </div>
          <input
            class="mastSearchInputCss"
            id="search"
            name="q"
            placeholder="Search..."
            type="text"
          />
          <input
            name="q"
            type="hidden"
            value="site:www.franksmovielog.com"
          />
          <button
            aria-label="Search"
            class="mastSearchSubmitCss"
            type="submit"
            value="Search"
          >
            <svg
              class="iconCss "
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10.442 10.442a1 1 0 011.415 0l3.85 3.85a1 1 0 01-1.414 1.415l-3.85-3.85a1 1 0 010-1.415z"
                fill-rule="evenodd"
              />
              <path
                d="M6.5 12a5.5 5.5 0 100-11 5.5 5.5 0 000 11zM13 6.5a6.5 6.5 0 11-13 0 6.5 6.5 0 0113 0z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </label>
      </form>
      <nav
        class="mastNavCss"
      >
        <ul
          class="mastNavListCss"
        >
          <li
            class="mastNavListItemCss"
          >
            <a
              class="mastNavLinkCss"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            class="mastNavListItemCss"
          >
            <a
              class="mastNavLinkCss"
              href="/about/"
            >
              About
            </a>
          </li>
          <li
            class="mastNavListItemCss"
          >
            <a
              class="mastNavLinkCss"
              href="/how-i-grade/"
            >
              How I Grade
            </a>
          </li>
          <li
            class="mastNavListItemCss"
          >
            <a
              class="mastNavLinkCss"
              href="/reviews/"
            >
              All Reviews
            </a>
          </li>
          <li
            class="mastNavListItemCss"
          >
            <a
              class="mastNavLinkCss"
              href="/viewings/"
            >
              Viewing Log
            </a>
          </li>
          <li
            class="mastNavListItemCss"
          >
            <a
              class="mastNavLinkCss"
              href="/watchlist/"
            >
              Watchlist
            </a>
          </li>
        </ul>
      </nav>
    </header>
    <div
      class="childrenCss"
      id="content"
    >
      <main
        class="containerCss"
      >
        <div
          class="leftCss"
        >
          <header
            class="pageHeaderCss"
          >
            <h2
              class="headingCss"
            >
              Viewing Log
            </h2>
            <p
              class="taglineCss"
            >
              I've watched 4 movies since 2012. 
              <span
                class="quoteCss"
              >
                “We have such sights to show you.”
              </span>
              <br />
              <a
                class="linkCss"
                href="/viewings/stats/"
              >
                Stats 
                <svg
                  class="iconCss"
                  fill="none"
                  stroke="currentColor"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M13 7h8m0 0v8m0-8l-8 8-4-4-6 6"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </a>
            </p>
          </header>
          <fieldset
            class="fieldsetCss filtersCss"
          >
            <legend>
              Filter & Sort
            </legend>
            <label
              class="labelCss"
              for="viewings-title-input"
            >
              Title
              <input
                class="inputCss "
                id="viewings-title-input"
                placeholder="Enter all or part of a title"
                type="text"
              />
            </label>
            <div
              class="containerCss"
              id="viewings-release-year-input"
            >
              <div
                id="viewings-release-year-input-label"
              >
                Release Year
              </div>
              <div
                class="sliderCss"
                style="position: relative;"
              >
                <div
                  class="trackCss trackCss-0"
                  style="position: absolute; left: 0px; right: 0px;"
                />
                <div
                  class="trackCss trackCss-1"
                  style="position: absolute; left: 0px; right: 0px;"
                />
                <div
                  class="trackCss trackCss-2"
                  style="position: absolute; left: 0px; right: 0px;"
                />
                <div
                  aria-orientation="horizontal"
                  aria-valuemax="2011"
                  aria-valuemin="1976"
                  aria-valuenow="1976"
                  class="thumbCss thumbCss-0 "
                  role="slider"
                  style="position: absolute; z-index: 1; left: 0px;"
                  tabindex="0"
                />
                <div
                  aria-orientation="horizontal"
                  aria-valuemax="2011"
                  aria-valuemin="1976"
                  aria-valuenow="2011"
                  class="thumbCss thumbCss-1 "
                  role="slider"
                  style="position: absolute; z-index: 2; left: 0px;"
                  tabindex="0"
                />
              </div>
              <input
                aria-labelledby="viewings-release-year-input-label"
                class="yearInputCss"
                max="2011"
                min="1976"
                step="1"
                type="number"
                value="1976"
              />
              <span
                class="toCss"
              >
                 to 
              </span>
              <input
                aria-labelledby="viewings-release-year-input-label"
                class="yearInputCss rightCss"
                max="2011"
                min="1976"
                step="1"
                type="number"
                value="2011"
              />
            </div>
            <label
              class="labelCss"
              for="viewings-venue-input"
            >
              Venue
              <select
                class="selectInputCss "
                id="viewings-venue-input"
              >
                <option
                  value="All"
                >
                  All
                </option>
                <option
                  value="AMC Tysons Corner 16"
                >
                  AMC Tysons Corner 16
                </option>
                <option
                  value="Arrow Player"
                >
                  Arrow Player
                </option>
                <option
                  value="Blu-ray"
                >
                  Blu-ray
                </option>
                <option
                  value="Hulu"
                >
                  Hulu
                </option>
              </select>
            </label>
            <label
              class="labelCss"
              for="viewings-sort-input"
            >
              Order By
              <select
                class="selectInputCss "
                id="viewings-sort-input"
              >
                <option
                  value="viewing-date-desc"
                >
                  Viewing Date (Newest First)
                </option>
                <option
                  value="viewing-date-asc"
                >
                  Viewing Date (Oldest First)
                </option>
                <option
                  value="release-date-desc"
                >
                  Release Date (Newest First)
                </option>
                <option
                  value="release-date-asc"
                >
                  Release Date (Oldest First)
                </option>
                <option
                  value="title"
                >
                  Title
                </option>
              </select>
            </label>
          </fieldset>
        </div>
        <div
          class="rightCss"
        >
          <ol
            class="listCss"
            data-testid="viewings-list"
          >
            <li
              class="listItemCss listItemFirstCss"
              value="1113"
            >
              <div
                class="listItemTitleCss"
              >
                <a
                  class="listItemTitleLinkCss"
                  href="/reviews/weekend-at-bernies-1989/#1113"
                  rel="canonical"
                >
                  Weekend at Bernie's 
                  <span
                    class="listItemTitleYearCss"
                  >
                    1989
                  </span>
                </a>
              </div>
              <div
                class="listItemSlugCss"
              >
                Saturday Jun 12, 2021 via Hulu
              </div>
            </li>
            <li
              class="listItemCss "
              value="1112"
            >
              <div
                class="listItemTitleCss"
              >
                <a
                  class="listItemTitleLinkCss"
                  href="/reviews/the-human-tornado-1976/#1112"
                  rel="canonical"
                >
                  The Human Tornado 
                  <span
                    class="listItemTitleYearCss"
                  >
                    1976
                  </span>
                </a>
              </div>
              <div
                class="listItemSlugCss"
              >
                Thursday Jun 10, 2021 via Blu-ray
              </div>
            </li>
            <li
              class="listItemCss "
              value="1111"
            >
              <div
                class="listItemTitleCss"
              >
                <a
                  class="listItemTitleLinkCss"
                  href="/reviews/gone-with-the-pope-2010/#1111"
                  rel="canonical"
                >
                  Gone with the Pope 
                  <span
                    class="listItemTitleYearCss"
                  >
                    2010
                  </span>
                </a>
              </div>
              <div
                class="listItemSlugCss"
              >
                Thursday Jun 10, 2021 via Arrow Player
              </div>
            </li>
            <li
              class="listItemCss "
              value="1"
            >
              <div
                class="listItemTitleCss"
              >
                The Girl with the Dragon Tattoo 
                <span
                  class="listItemTitleYearCss"
                >
                  2011
                </span>
              </div>
              <div
                class="listItemSlugCss"
              >
                Thursday Jan 5, 2012 via AMC Tysons Corner 16
              </div>
            </li>
          </ol>
        </div>
      </main>
    </div>
    <footer
      class="footerCss"
    >
      <ul
        class="footerNavListCss"
      >
        <li
          class="footerNavListItemCss"
        >
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="footerNavListItemCss"
        >
          <a
            href="/about/"
          >
            About
          </a>
        </li>
        <li
          class="footerNavListItemCss"
        >
          <a
            href="/how-i-grade/"
          >
            How I Grade
          </a>
        </li>
        <li
          class="footerNavListItemCss"
        >
          <a
            href="/reviews/"
          >
            All Reviews
          </a>
        </li>
        <li
          class="footerNavListItemCss"
        >
          <a
            href="/viewings/"
          >
            Viewing Log
          </a>
        </li>
        <li
          class="footerNavListItemCss"
        >
          <a
            href="/watchlist/"
          >
            Watchlist
          </a>
        </li>
      </ul>
      <p
        class="footerFairUseCss"
      >
        All stills used in accordance with the 
        <a
          href="http://www.copyright.gov/title17/92chap1.html#107"
        >
          Fair Use Law.
        </a>
      </p>
      <div
        class="srOnlyCss"
      >
        <a
          href="#top"
        >
          To the top ↑
        </a>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
